symphony/bdk/core/retry/strategy.py

```diff
@@ -3,41 +3,70 @@
+        return exception.status == 401
+    return False
+
+
+def is_client_error(exception: Exception) -> bool:
+    if isinstance(exception, ApiException):
+        return exception.status == 400
+    return False
+
+
+def is_client_timeout_error(exception: Exception):
+    """Checks if the exception is a :py:class:`ClientConnectorError` with a :py:class:`TimeoutError` as cause
+
+    :param exception: The exception to be checked
+    :return: True if checks the predicate, False otherwise
+    """
+    return isinstance(exception, ClientConnectorError) and isinstance(exception.__cause__, TimeoutError)
 
 
-def is_client_error(exception: ApiException) -> bool:
-    return exception.status == 400
+def can_authentication_be_retried(exception: Exception) -> bool:
+    """Checks if the exception is a network issue or internal or retry error
+    i.e. is an :py:class:`ApiException` with status code 429 or greater or equal to 500.
+    or the exception is either a :py:cass:`TimeoutError`  (e.g. network issues)
+
+    :param exception: The exception to be checked
+    :return: True if the exception is an :py:class:`ApiException` with status 500, 429 or or is client timeout error
+    False otherwise
+    """
+    if isinstance(exception, ApiException):
+        return exception.status >= 500 or exception.status == 429
+    return is_client_timeout_error(exception)
 
 
 def is_network_or_minor_error(exception: Exception) -> bool:
-    """Checks if the exception is a network issue or an :py:class:`ApiException` minor error
+    """Checks if the exception is a network issue or an :py:class:`ApiException` minor error (internal + unauthorized + retry)
     This is the default function used to check if a given exception should lead to a retry
 
     :param exception: The exception to be checked
-    :return:
+    :return: True if the exception is an :py:class:`ApiException` with status 500, 401, 429 or is a client timeout error
+    False otherwise
     """
     if isinstance(exception, ApiException):
         if exception.status >= 500 or exception.status == 401 or exception.status == 429:
             return True
-    return isinstance(exception, ClientConnectorError) and isinstance(exception.__cause__, TimeoutError)
+    return is_client_timeout_error(exception)
 
 
-def can_authentication_be_retried(exception: ApiException) -> bool:
-    """Predicate to check if authentication call be retried,
-    i.e. is an :py:class:`ApiException` with status code 429 or greater or equal to 500.
-    or the exception is either a :py:cass:`TimeoutError`  (e.g. network issues)
+def is_network_or_minor_error_or_client(exception: Exception) -> bool:
+    """Checks if the exception is a network issue or an :py:class:`ApiException` minor error or a client error
 
     :param exception: The exception to be checked
-    :return:
+    :return: True if the exception is an :py:class:`ApiException` with status 500, 401, 429, 400
+    or is a client timeout error False otherwise
```

Comment:
Beware the indentation for the second line of the `:return:` part!

Reply:
I looked it up on the internet but didn't really get it ? Do you want to add an indent or align it with the previous line (2 indents and a space)